### ISSUE TITLE
Move from VATRates to node-sales-tax

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#3570]
Following https://github.com/staaky/vatrates/issues/9#issuecomment-643457104, I've been talking with people about VATRates being discontinued and we ended up looking at https://github.com/valeriansaliou/node-sales-tax. This library is well more maintained, and aims to provide support for all taxes worldwide - which is great now that we decided to support GST as well.

The main challenge with this library is that it uses async calls, mostly to validate the tax ID numbers against public APIs.

In short, node-sales-tax would bring the following benefits:
- Simpler code. We may event want to remove `opencollective-taxes` at some point if it works as promised.
- Maintained (VATRates is not), which also means up-to-date rates
- Support for more taxes out of the box
